viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
+            avg_rating = 0.0
+        else:
+            avg_rating = total_rating / len(user_data['watched'])
+    return avg_rating
+            
+
 
 
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    if user_data['watched'] == []:
+        return None
+    for movie_dict in user_data['watched']:
+        current_genre = movie_dict['genre']
+        if current_genre not in genre_count:
+            genre_count[current_genre] = 1
+        else:
+            genre_count[current_genre] += 1
+    genre_most_watched = max(genre_count, key = genre_count.get)
+    return genre_most_watched
+    
+    
+        
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_friends_movies(user_data):
+    friends_movies = []
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            friends_movies.append(movie)
+    return friends_movies
+
+
+def get_unique_watched(user_data):
+    unique_movies = []
+    friends_movies = get_friends_movies(user_data)
+    for movie_dict in user_data['watched']:
+        if movie_dict not in friends_movies:
+            unique_movies.append(movie_dict)
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    friends_unique_movies = []
+    index = 0
+    for movies_dict in user_data['watched']:
+        user_watched_list.append(movies_dict['title'])
+
+    for friends_dict in user_data['friends']:
+        movie_list = friends_dict['watched']
+        for movies_dict in movie_list:
+            if movies_dict in friends_unique_movies:
+                continue
+            if movies_dict['title'] not in user_watched_list:
+                friends_unique_movies.append(movies_dict)
+    return friends_unique_movies
+
+    
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    friends_movies = get_friends_unique_watched(user_data)
+    subscriptions = user_data['subscriptions']
+    reccomendations = []
+
+    for movie_dict in friends_movies:
+        if movie_dict['host'] in subscriptions:
+            reccomendations.append(movie_dict)
+    return reccomendations
+
+
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+    recommended_movies = []
+    friends_unique_movies = get_friends_unique_watched(user_data)
+    for movie in friends_unique_movies:
+        if movie['genre'] == most_watched_genre:
+            recommended_movies.append(movie)
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):
+    user_favorites = user_data['favorites'] #list of dictionaries

```

Comment:
Instead of leaving a comment about what the variable is, we should try to name the variable in a descriptive way that enables to remember what returns from `user_data['favorites']`, maybe something like `user_favorites_list`